src/dtypes.rs

```diff
@@ -3,12 +3,18 @@ use crate::kzg_proof::safe_scalar_affine_from_bytes;
 use crate::{BYTES_PER_BLOB, BYTES_PER_FIELD_ELEMENT};
 use alloc::{string::ToString, vec::Vec};
 use bls12_381::Scalar;
-use serde::{Deserialize, Serialize};
 
 macro_rules! define_bytes_type {
     ($name:ident, $size:expr) => {
-        #[derive(Debug, Clone, Serialize, Deserialize)]
-        pub struct $name(#[serde(with = "serde_arrays")] pub [u8; $size]);
+        #[cfg_attr(
+            feature = "rkyv",
+            derive(rkyv::Archive, rkyv::Serialize, rkyv::Deserialize)
+        )]
+        #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+        #[derive(Debug, Clone)]
+        pub struct $name(
+            #[cfg_attr(feature = "serde", serde(with = "serde_arrays"))] pub [u8; $size],
```

Comment:
does this work with bincode?
curious if a roundtrip with this works ~ i was previously getting stack overflow errors using bincode + serde (serde_big_array / serde_bytes)